### PR TITLE
16 sidebar related news

### LINKED ISSUE #16: Sidebar Related Content auto-block
Implement an auto-block able to fetch related content based on a defined template and index, for example:
Mack News:

- template --> mack-news
- index --> mack-news.json

image

### PR DESCRIPTION
**Pull request by TomaszDziezykNetcentric**
Fix #16 

Test URLs:
- Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/drafts/manuel.vara/mack-news/2021/mack-displays-breadth-of-mack-granite-model-lineup-with-augmented-reality
- After: https://16-sidebar-related-news--vg-macktrucks-com--hlxsites.hlx.page/drafts/manuel.vara/mack-news/2021/mack-displays-breadth-of-mack-granite-model-lineup-with-augmented-reality

### CODE REVIEW

**Review by jonatan-lledo-netcentric — CHANGES_REQUESTED** [review]

File: scripts/scripts.js [hunk 1]
```diff
@@ -27,6 +75,77 @@ function buildHeroBlock(main) {
   }
 }
 
+/**
+ * loads more data from the query index
+ * */
+async function loadMoreNews() {
+  if (!window.mack.newsData.allLoaded) {
+    const queryLimit = 200;
+    /*
```
[the remark this answers is not in the document]
Comment (TomaszDziezykNetcentric) [reply scripts/scripts.js]: Commented code removed

File: blocks/fragment/fragment.js [hunk 2]
```diff
@@ -37,7 +37,10 @@ export default async function decorate(block) {
   const fragment = await loadFragment(path);
   if (fragment) {
     const fragmentSection = fragment.querySelector(':scope .section');
+    const hr = document.createElement('hr');
```
Comment (manuel-vara) [inline blocks/fragment/fragment.js:40]: Are we sure we absolutely need this one? Please be aware this block can be reused on every content page, so it should remain as generic as possible.

Comment (TomaszDziezykNetcentric) [reply blocks/fragment/fragment.js]: I removed the `hr` element from `fragment.js`.
I added some CSS for the fragment when it is on the mack news page only.

BTW I change the 
```
block.closest('.section').classList.add(...fragmentSection.classList);
block.closest('.fragment-wrapper').replaceWith(...fragmentSection.childNodes);
```
with
```
block.classList.add(...fragmentSection.classList, 'fragment');
block.replaceChildren(...fragmentSection.childNodes);
```
before the classes were added to the news section but it should be added to the fragment block
From the user perspective it looks the same.

Comment (manuel-vara) [reply blocks/fragment/fragment.js]: Thanks

**Comment by manuel-vara** [comment]
Can you have a look to pagespeed insights report and try to implement some of the opportunities for improvement of the speed index, please?

**Comment by TomaszDziezykNetcentric** [comment]
> Can you have a look to pagespeed insights report and try to implement some of the opportunities for improvement of the speed index, please?

Now it should be fine

**Review by manuel-vara — APPROVED** [review]

**Pull request merged**